mica/starcheck/starcheck.py

```diff
@@ -169,211 +173,125 @@ def get_starcheck_catalog_at_date(date, starcheck_db=None, timelines_db=None):
-    # nmm sequence like dark cal.
-
-    # There is a tiny window of time in cmd_states but not yet in kadi, but this code tries to
-    # grab the dwell and maneuver that would be related to a date in that range
-    if date < last_tl and len(dwells) == 0:
-        pcad_states = get_states(
-            start=DateTime(date) - 2, state_keys=["pcad_mode"], merge_identical=True
-        )
-        dwell = pcad_states[
-            (pcad_states["pcad_mode"] == "NPNT") & (pcad_states["datestop"] >= date)
-        ][0]
-        manvr = pcad_states[
-            (pcad_states["pcad_mode"] == "NMAN")
-            & (pcad_states["datestop"] <= dwell["datestart"])
-        ][-1]
-        start_cat_search = manvr["datestart"]
-        dwell_start = dwell["datestart"]
-    else:
-        # If we have a dwell from kadi, use it to search for commanding
-        dwell = dwells[0]
-        dwell_start = dwell.start
-        # Try to use the beginning of the previous nman period to define when the catalog
-        # should have been commanded.  If there are multiple dwells for attitude, try to
-        # use nman_start if available.
-        if dwell.manvr.n_dwell > 1 and dwell.manvr.nman_start is not None:
-            start_cat_search = dwell.manvr.nman_start
-        else:
-            start_cat_search = dwell.get_previous().stop
-
-    timelines = timelines_db.fetchall(
-        """select * from timeline_loads where scs < 131
-           and datestop > '{}' and datestart < '{}' order by datestart""".format(
-            start_cat_search, dwell_start
-        )
-    )
-    for timeline in timelines[::-1]:
-        starchecks = db.fetchall(
-            """select * from starcheck_obs, starcheck_id
-               where dir = '{}'
-               and mp_starcat_time >= '{}'
-               and mp_starcat_time <= '{}' and mp_starcat_time <= '{}'
-               and starcheck_id.id = starcheck_obs.sc_id
-               order by mp_starcat_time """.format(
-                timeline["mp_dir"],
-                timeline["datestart"],
-                timeline["datestop"],
-                dwell_start,
-            )
-        )
-        # The last one should be the one before beginning of the dwell
-        if len(starchecks):
-            # Use the obsid and the known products directory to use the more
-            # generic get_starcheck_catalog to fetch the right one from the
-            # database
-            cat_info = get_starcheck_catalog(
-                starchecks[-1]["obsid"], mp_dir=starchecks[-1]["dir"]
-            )
-            cat_info["status"] = "ran" if date < DateTime().date else "approved"
-            return cat_info
-    return None
 
+    # Get observations within +/- 7 days and find the one where ``date`` is between end
+    # of the previous obs dwell through the end of the current obs dwell.
+    obss = Table(get_observations(start=date - 7 * u.day, stop=date + 7 * u.day))
```

Comment:
I think this move to kadi observations is fine, but does result in *different* results, such as for the high IR gyro holds.  I think this is can be handled maybe with just an update to the docstring (which was poor before anyway).

Reply:
Example.  Old mica
```
In [26]: mica.starcheck.get_starcheck_catalog_at_date('2023:099:04:21:40.719')
Out[26]: 
{'mp_dir': '/2023/APR0323/oflsa/',
 'status': 'approved',
 'obs': {'sc_id': 2477,
  'obsid': 44652,
  'obs_idx': 46,
  'point_ra': 304.0,
  'point_dec': -52.0,
  'point_roll': 107.697667,
  'target_id': None,
  'sci_instr': None,
  'sim_z_offset_steps': None,
  'sim_z_offset_mm': None,
  'grating': None,
  'dither_state': 'ON',
  'dither_y_amp': 8.0,
  'dither_y_period': 1000.0,
  'dither_z_amp': 8.0,
  'dither_z_period': 707.1,
  'mp_starcat_time': '2023:099:05:01:44.468',
  'mp_starcat_vcdu_cnt': 442966},
 'manvr': <Table length=1>
 sc_id obsid obs_idx instance ... target_Q4     mp_targquat_time   mp_targquat_vcdu_cnt      end_date    
 int64 int64  int64   int64   ...  float64           str21                int64               str17      
 ----- ----- ------- -------- ... ---------- --------------------- -------------------- -----------------
  2477 44652      46        0 ... 0.30197339 2023:099:05:01:42.825               442960 2023:099:05:05:38,
 'warnings': [],
 'cat': <Table length=8>
 sc_id obsid obs_idx    mp_starcat_time     idx   slot     id     ...  yang  zang  dim   res  halfw pass notes
 int64 int64  int64          str21         int64 int64   int64    ... int64 int64 int64 int64 int64 str2  str2
 ----- ----- ------- --------------------- ----- ----- ---------- ... ----- ----- ----- ----- ----- ---- -----
  2477 44652      46 2023:099:05:01:44.468     1     0 1103236736 ...  -167 -2381    28     1   160   --    --
  2477 44652      46 2023:099:05:01:44.468     2     1 1103233664 ...  1232 -1987    28     1   160   --    --
  2477 44652      46 2023:099:05:01:44.468     3     2 1103241832 ...  1473 -1659    24     1   140   --    --
  2477 44652      46 2023:099:05:01:44.468     4     3 1103240512 ...  1636 -1067    28     1   160   --    --
  2477 44652      46 2023:099:05:01:44.468     5     4 1150420232 ... -1964   320    28     1   160   --    --
  2477 44652      46 2023:099:05:01:44.468     6     5 1101531176 ...  -836  1697    28     1   160   --    --
  2477 44652      46 2023:099:05:01:44.468     7     6 1103240096 ...  1858 -2397    28     1   160   --    --
  2477 44652      46 2023:099:05:01:44.468     8     7 1103233952 ...  -649  -851    28     1   160   --    --,
 'pred_temp': -10.8}
```

PR testing
```
In [3]: mica.starcheck.get_starcheck_catalog_at_date('2023:099:04:21:40.719')
Out[3]: 
{'mp_dir': '/2023/APR0323/oflsa/',
 'status': 'approved',
 'obs': {'sc_id': 2477,
  'obsid': 44653,
  'obs_idx': 45,
  'point_ra': 304.0,
  'point_dec': -52.0,
  'point_roll': 107.709085,
  'target_id': None,
  'sci_instr': None,
  'sim_z_offset_steps': None,
  'sim_z_offset_mm': None,
  'grating': None,
  'dither_state': None,
  'dither_y_amp': None,
  'dither_y_period': None,
  'dither_z_amp': None,
  'dither_z_period': None,
  'mp_starcat_time': None,
  'mp_starcat_vcdu_cnt': None},
 'manvr': [],
 'warnings': [],
 'cat': []}
```

Reply:
Fundamentally, it was never clear what a "starcheck catalog" was, to get it with this function.  For this example, the new behavior seems also fine -- it gets the starcheck entry for a field without a star catalog.

Reply:
Nice example, and I would argue that this is fixing an unexpected behavior in the current mica.  As you note there is no hard spec on the correct answer, but I definitely expect the function to return the observation dwell which contains the  `date` argument. If that dwell has no catalog then that is the answer.